### ISSUE TITLE
Fix output of `iota validator make-validator-info`

### ISSUE THREAD

**Issue by alexsporn** [iotaledger/iota#4437]
When running `iota validator make-validator-info` the generated validator.info contains these lines:

```
...
  network-address: /dns/65.109.144.22/tcp/8080/http
  p2p-address: /dns/65.109.144.22/udp/8084
  primary-address: /dns/65.109.144.22/udp/8081
...
```

There are multiple issues that should be fixed:

1. The primary-address should use `.../tcp/8081`
2. If the passed `<HOST_NAME>` argument is an IPv4, we should use `/ip4/65.109.144.22/...` instead

None of these issues really cause stuff not to work, but it would reduce confusion when setting up a validator.

**Comment by gokhan-simsek-iota** [comment]
Started with `primary-address`, I see that Sui says `/tcp/8081` in their doc. 
https://docs.sui.io/guides/operator/validator-tasks#connectivity , but in their code it is still `udp/8081` everywhere, except in `narwhal/config/tests`.

I had the impression that this is just a small change, looks like it is: https://github.com/iotaledger/iota/blob/cf7ec2bd6de6bf6054e53ea718c9a61fe5d39a40/crates/iota/src/validator_commands.rs#L347-L348 

but I see that to be consistent we need to change many other parts of the code from `udp` to `tcp`:
```
search "/udp/8081"
9 results - 3 files

docker/iota-network/genesis/compose-validators.yaml:
  19      name: validator1
  20:     primary_address: /dns/validator1/udp/8081

  28      name: validator2
  29:     primary_address: /dns/validator2/udp/8081
...

docker/iota-private-network/configs/genesis-template.yaml:
...

docs/content/references/cli/validator.mdx:
  181    "p2pAddress": "/dns/validator-2.r.iota-rebased-testnet.iota.cafe/udp/8084",
  182:   "primaryAddress": "/dns/validator-2.r.iota-rebased-testnet.iota.cafe/udp/8081",
  183    "nextEpochAuthorityPubkeyBytes": null,
```
```
search "primary_address"
118 results - 25 files

e.g.
crates/iota/src/generate_genesis_checkpoint.rs:
  38:             primary_address: local_ip_utils::new_local_udp_address_for_testing(),
```

**Comment by alexsporn** [comment]
@gokhan-simsek-iota yes, you can go ahead and update all references to 8081/udp to be 8081/tcp. Those are still from the Narwhal consensus.

**Comment by gokhan-simsek-iota** [comment]
**Update:**
After latest changes and fixing some of the failing CI checks, we now have new failing 72  `iota-graphql-e2e-tests` failing tests and this one `iota-e2e-tests::snapshot_tests basic_read_cmd_snapshot_tests` left.

**Next:** 
- Find how to decide if the new exp file makes sense. 
   (Seems like `iota-graphql-e2e-tests` are failing due to un-updated `exp` files, I found a way to update the `exp` files but there are too many diffs and very difficult for me to decide if the new `exp` file makes sense, some of the changes are not just digest changes.)
- Find a way to update `crates/iota-e2e-tests/tests/snapshots/snapshot_tests__body_fn.snap`, `scripts/update_all_snapshots.sh` did not help

**Details:**

Failing `iota-graphql-e2e-tests` tests:
```
------------
     Summary [  69.449s] 85 tests run: 13 passed, 72 failed, 0 skipped
        FAIL [   4.313s] iota-graphql-e2e-tests::tests run_test::available_range/available_range.move
        FAIL [   5.298s] iota-graphql-e2e-tests::tests run_test::call/dynamic_fields.move
        ...
```

Failing `iota-e2e-tests::snapshot_tests basic_read_cmd_snapshot_tests` test:
```
iota-e2e-tests::snapshot_tests basic_read_cmd_snapshot_tests
			━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
			Snapshot file: crates/iota-e2e-tests/tests/snapshots/snapshot_tests__body_fn.snap
			Snapshot: body_fn
			Source: crates/iota-e2e-tests/tests/snapshot_tests.rs:73
```

**Comment by miker83z** [comment]
I see that #4753 targets protocol version 4. Couldn't it be possible to just create a PR for a preliminary step where the constraints to the `make-validator-info` are removed and use 3 MultiAddresses like in [CeremonyCommand::AddValidator](https://github.com/iotaledger/iota/blob/d7eb3902e725fe8ac2fc1332ca0cc3908d661d41/crates/iota/src/genesis_ceremony.rs#L184) `network_address: Multiaddr, p2p_address: Multiaddr, primary_address: Multiaddr`?

@gokhan-simsek-iota @alexsporn

**Comment by muXxer** [comment]
@miker83z what do you exactly mean? to remove the checks if it's a UDP or TCP MultiAddr?
The network layer only supports TCP so UDP is actually misleading.

**Comment by miker83z** [comment]
@muXxer 
> [@miker83z](https://github.com/miker83z) what do you exactly mean? to remove the checks if it's a UDP or TCP MultiAddr? The network layer only supports TCP so UDP is actually misleading.

I mean, even before waiting for Protocol Version 4, it would be nice to have `make-validator-info` without string validation (like in `genesis-ceremony add-validator`) so that I don't have to fix the validator.info file manually.

**Comment by gokhan-simsek-iota** [comment]
Latest updates/todos will be in this PR's comment: https://github.com/iotaledger/iota/pull/5919#issue-2917993640

**Comment by alexsporn** [comment]
Removed from protocol v5 milestone for now
